clang-23 warning with ArrayView constructor.

clang-23 gives me a warning with which I don't really know what to do:
```
/tempest/a/accounts/bangerth/p/deal.II/1/dealii/include/deal.II/base/array_view.h:549:49: warning: definition of implicit copy assignment operator for 'ArrayView<std::pair<unsigned long, int>>' is deprecated because it has a user-provided copy constructor [-Wdeprecated-copy-with-user-provided-copy]
  549 | inline ArrayView<ElementType, MemorySpaceType>::ArrayView(
      |                                                 ^
/usr/lib/gcc/x86_64-linux-gnu/12/../../../../include/c++/12/bits/stl_tempbuf.h:216:16: note: in implicit copy assignment operator for 'dealii::ArrayView<std::pair<unsigned long, int>>' first required here
  216 |               *__seed = _GLIBCXX_MOVE(*__prev);
      |                       ^
/usr/lib/gcc/x86_64-linux-gnu/12/../../../../include/c++/12/bits/stl_tempbuf.h:254:4: note: in instantiation of function template specialization 'std::__uninitialized_construct_buf_dispatch<false>::__ucr<dealii::ArrayView<std::pair<unsigned long, int>> *, std::_Deque_iterator<dealii::ArrayView<std::pair<unsigned long, int>>, dealii::ArrayView<std::pair<unsigned long, int>> &, dealii::ArrayView<std::pair<unsigned long, int>> *>>' requested here
  254 |           __ucr(__first, __last, __seed);
      |           ^
/usr/lib/gcc/x86_64-linux-gnu/12/../../../../include/c++/12/bits/stl_tempbuf.h:269:13: note: in instantiation of function template specialization 'std::__uninitialized_construct_buf<dealii::ArrayView<std::pair<unsigned long, int>> *, std::_Deque_iterator<dealii::ArrayView<std::pair<unsigned long, int>>, dealii::ArrayView<std::pair<unsigned long, int>> &, dealii::ArrayView<std::pair<unsigned long, int>> *>>' requested here
  269 |               std::__uninitialized_construct_buf(__p.first, __p.first + __p.second,
      |                    ^
/usr/lib/gcc/x86_64-linux-gnu/12/../../../../include/c++/12/bits/stl_algo.h:4996:15: note: in instantiation of member function 'std::_Temporary_buffer<std::_Deque_iterator<dealii::ArrayView<std::pair<unsigned long, int>>, dealii::ArrayView<std::pair<unsigned long, int>> &, dealii::ArrayView<std::pair<unsigned long, int>> *>, dealii::ArrayView<std::pair<unsigned long, int>>>::_Temporary_buffer' requested here
 4996 |       _TmpBuf __buf(__first, (__last - __first + 1) / 2);
      |               ^
/usr/lib/gcc/x86_64-linux-gnu/12/../../../../include/c++/12/bits/stl_algo.h:5070:23: note: in instantiation of function template specialization 'std::__stable_sort<std::_Deque_iterator<dealii::ArrayView<std::pair<unsigned long, int>>, dealii::ArrayView<std::pair<unsigned long, int>> &, dealii::ArrayView<std::pair<unsigned long, int>> *>, __gnu_cxx::__ops::_Iter_comp_iter<(lambda at /tempest/a/accounts/bangerth/p/deal.II/1/dealii/source/grid/grid_in.cc:4517:26)>>' requested here
 5070 |       _GLIBCXX_STD_A::__stable_sort(__first, __last,
      |                       ^
/tempest/a/accounts/bangerth/p/deal.II/1/dealii/source/grid/grid_in.cc:4515:14: note: in instantiation of function template specialization 'std::stable_sort<std::_Deque_iterator<dealii::ArrayView<std::pair<unsigned long, int>>, dealii::ArrayView<std::pair<unsigned long, int>> &, dealii::ArrayView<std::pair<unsigned long, int>> *>, (lambda at /tempest/a/accounts/bangerth/p/deal.II/1/dealii/source/grid/grid_in.cc:4517:26)>' requested here
 4515 |         std::stable_sort(face_id_to_sideset_ids.begin(),
      |              ^
/tempest/a/accounts/bangerth/p/deal.II/1/dealii/source/grid/grid_in.cc:4763:15: note: in instantiation of function template specialization 'dealii::(anonymous namespace)::read_exodusii_sidesets<1, 1>' requested here
 4763 |   auto pair = read_exodusii_sidesets<dim, spacedim>(
      |               ^
1 warning generated.
```
Anyone have an idea what it is trying to say?

--- a/include/deal.II/base/array_view.h
+++ b/include/deal.II/base/array_view.h
@@ -131,16 +131,27 @@ public:
   ArrayView(value_type *starting_element, const std::size_t n_elements);
 
   /**
-   * Copy constructor from array views that point to non-@p const elements. If
-   * the current object will point to non-@p const elements, then this is a
-   * straight forward copy constructor. On the other hand, if the current
-   * type's @p ElementType template argument is a @p const qualified type,
-   * then the current constructor is a conversion constructor that converts a
-   * non-@p const view to a @p const view, akin to converting a non-@p const
-   * pointer to a @p const pointer.
+   * Copy constructor.
    */
-  ArrayView(
-    const ArrayView<std::remove_cv_t<value_type>, MemorySpaceType> &view);
+  ArrayView(const ArrayView &view) = default;
+
+  /**
+   * Copy constructor from array views that point to non-@p const elements.
+   *
+   * This constructor only exists if the current ArrayView object
+   * points to `const` elements and the one given as argument points
+   * to non-`const` elements. If both point to `const` or non-`const`
+   * ones, the straightforward copy-constructor above is used. If the
+   * current object would point to non-`const` elements but the given
+   * argument points to `const` elements, the conversion is clearly
+   * not valid because it would remove the `const`ness of the
+   * argument.
+   */
+  template <typename OtherElementType,
+            typename = std::enable_if_t<
+              std::is_same_v<ElementType, const OtherElementType> &&
+              !std::is_same_v<OtherElementType, const ElementType>>>
+  ArrayView(const ArrayView<OtherElementType, MemorySpaceType> &view);
 
   /**
    * A constructor that automatically creates a view from a single value_type
@@ -318,6 +329,31 @@ public:
   ArrayView(
     const std::initializer_list<std::remove_cv_t<value_type>> &initializer_list)
     DEAL_II_CXX20_REQUIRES(std::is_const_v<ElementType>);
+
+  /**
+   * Copy operator.
+   */
+  ArrayView<ElementType, MemorySpaceType> &
+  operator=(const ArrayView &view) = default;
+
+  /**
+   * Copy operator from array views that point to non-@p const elements.
+   *
+   * This operator only exists if the current ArrayView object
+   * points to `const` elements and the one given as argument points
+   * to non-`const` elements. If both point to `const` or non-`const`
+   * ones, the straightforward copy-constructor above is used. If the
+   * current object would point to non-`const` elements but the given
+   * argument points to `const` elements, the conversion is clearly
+   * not valid because it would remove the `const`ness of the
+   * argument.
+   */
+  template <typename OtherElementType,
+            typename = std::enable_if_t<
+              std::is_same_v<ElementType, const OtherElementType> &&
+              !std::is_same_v<OtherElementType, const ElementType>>>
+  ArrayView<ElementType, MemorySpaceType>
+  operator=(const ArrayView<OtherElementType, MemorySpaceType> &view);
 
   /**
    * Reinitialize a view.
@@ -546,8 +582,9 @@ inline ArrayView<ElementType, MemorySpaceType>::ArrayView(ElementType &element)
 
 
 template <typename ElementType, typename MemorySpaceType>
+template <typename OtherElementType, typename>
 inline ArrayView<ElementType, MemorySpaceType>::ArrayView(
-  const ArrayView<std::remove_cv_t<value_type>, MemorySpaceType> &view)
+  const ArrayView<OtherElementType, MemorySpaceType> &view)
   : starting_element(view.starting_element)
   , n_elements(view.n_elements)
 {}
@@ -622,6 +659,19 @@ inline ArrayView<ElementType, MemorySpaceType>::ArrayView(
   : // use delegating constructor
   ArrayView(initializer.begin(), initializer.size())
 {}
+
+
+
+template <typename ElementType, typename MemorySpaceType>
+template <typename OtherElementType, typename>
+ArrayView<ElementType, MemorySpaceType>
+ArrayView<ElementType, MemorySpaceType>::operator=(
+  const ArrayView<OtherElementType, MemorySpaceType> &view)
+{
+  starting_element = view.starting_element;
+  n_elements       = view.n_elements;
+  return *this;
+}
 
 
 

